anchors/service.go

```diff
@@ -2,132 +2,64 @@ package anchors
-func (s *service) HasValidPreCommit(anchorID AnchorID) bool {
-	opts, cancelF := s.client.GetGethCallOpts(false)
-	defer cancelF()
-	r, err := s.anchorRepositoryContract.HasValidPreCommit(opts, anchorID.BigInt())
-	if err != nil {
-		return false
-	}
-	return r
+func newService(config Config, anchorRepository Repository, queue *queue.Server, client centchain.API, jobsMan jobs.Manager) AnchorRepository {
+	return &service{config: config, anchorRepository: anchorRepository, client: client, queue: queue, jobsMan: jobsMan}
 }
 
 // GetAnchorData takes an anchorID and returns the corresponding documentRoot from the chain.
 // Returns a nil error when the anchor data is found else returns a non nil error
 func (s *service) GetAnchorData(anchorID AnchorID) (docRoot DocumentRoot, anchoredTime time.Time, err error) {
-	opts, cancelF := s.client.GetGethCallOpts(false)
-	defer cancelF()
-	r, err := s.anchorRepositoryContract.GetAnchorById(opts, anchorID.BigInt())
+	r, err := s.anchorRepository.GetAnchorByID(anchorID.BigInt())
 	if err != nil {
 		return docRoot, anchoredTime, err
 	}
 
 	if utils.IsEmptyByte32(r.DocumentRoot) {
-		return docRoot, anchoredTime, errors.New("anchor data missing for id: %v", anchorID.String())
+		return docRoot, anchoredTime, errors.New("anchor data empty for id: %v", anchorID.String())
 	}
 
-	blk, err := s.client.GetBlockByNumber(context.Background(), big.NewInt(int64(r.BlockNumber)))
+	//TODO get block time
+	//blk, err := s.client.GetBlockByNumber(context.Background(), big.NewInt(int64(r.BlockNumber)))
+	//if err != nil {
+	//	return docRoot, anchoredTime, err
+	//}
+	bts, err := types.HexDecodeString(r.DocumentRoot.Hex())
 	if err != nil {
 		return docRoot, anchoredTime, err
 	}
-
-	return r.DocumentRoot, time.Unix(blk.Time().Int64(), 0), nil
-}
-
-// PreCommitAnchor will call the transaction PreCommit on the smart contract
-func (s *service) PreCommitAnchor(ctx context.Context, anchorID AnchorID, signingRoot DocumentRoot) (confirmations chan error, err error) {
-	did, err := getDID(ctx)
+	dr, err := ToDocumentRoot(bts)
 	if err != nil {
-		return nil, err
-	}
-
-	tc, err := contextutil.Account(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	jobID := contextutil.Job(ctx)
-
-	conn := s.client
-	opts, err := conn.GetTxOpts(ctx, tc.GetEthereumDefaultAccountName())
-	if err != nil {
-		return nil, err
-	}
-
-	opts.GasLimit = s.config.GetEthereumGasLimit(config.AnchorPreCommit)
-	pc := newPreCommitData(anchorID, signingRoot)
-	log.Infof("Add Anchor to Pre-commit %s from did:%s", anchorID.String(), did.ToAddress().String())
-	_, done, err := s.jobsMan.ExecuteWithinJob(contextutil.Copy(ctx), did, jobID, "Check Job for anchor commit",
-		s.ethereumTX(opts, s.anchorRepositoryContract.PreCommit, pc.AnchorID.BigInt(), pc.SigningRoot))
-	if err != nil {
-		return nil, err
+		return docRoot, anchoredTime, err
 	}
 
-	return done, nil
+	return dr, time.Unix(0, 0), nil
```

Comment:
are we not returning the actual anchor time here ?

Reply:
As I mentioned above in the comment, it is a todo item. We need a way to get the block time out of the block events? I didnt want to block this PR just for this.